Move prerequisites link into course info

MinhxNguyen7 opened the pull request:
## Summary
- [x] Removed the Prerequisites button
- [x] Made the "Prerequisites" text link to the prereqs page like the button does.
![image](https://user-images.githubusercontent.com/64875104/189587952-85da0566-ff29-4c2d-bb09-2121b7d8d3d5.png)
The link acts the same as the late prerequisites button, except that it opens the page in a new tab (which I think is standard since it leads to a different site).

## Test Plan
Check that the link leads to the prerequisites page in a new tab.
Check that the button has disappeared for good.

## Issues
Closes #393

EricPedley committed:
Add recruitment banner (#427)

MinhxNguyen7 committed:
Move prereq link to course info popover

MinhxNguyen7 committed:
Dismiss button for banner (#430)

* add recruitment banner

* Fix typo in recruitment banner

Co-authored-by: Eddy Chen <89349085+ecxyzzy@users.noreply.github.com>

* right-align banner and rewrite message

* Tooltip for Import button

* Dismiss button for banner

* Dismiss button for banner

* Recruitment banner border

* change "positions" to "opportunities"

* Extracted styles and moved bannerContainer div

* recruitment banner 11-week dismissal

* Revert "Tooltip for Import button"

Change snuck in here by accident.

* Change 6652800000 ms to representative expression

11 * 7 * 24 * 3600 * 1000 = 6652800000 = ms in a week.

Co-authored-by: Eric Pedley <ericpedley@gmail.com>
Co-authored-by: Eddy Chen <89349085+ecxyzzy@users.noreply.github.com>

MinhxNguyen7 committed:
Fix Kanban link (#436)

EricPedley commented on src/components/RightPane/SectionTable/SectionTable.js:
```diff
@@ -82,21 +82,16 @@ const SectionTable = (props) => {
                     courseNumber={courseDetails.courseNumber}
+                    prerequisiteAnchorInfo={{
+                        link: courseDetails.prerequisiteLink,
+                        analyticsAction: analyticsEnum.classSearch.actions.CLICK_PREREQUISITES,
```
is this value ever not `analyticsEnum.classSearch.actions.CLICK_PREREQUISITES`? If not, I don't think it needs to be a field in this object, it can just be hard-coded into CouseInfoBar.

EricPedley reviewed (changes requested):
1 comment to address

EricPedley committed:
Update README.md

MinhxNguyen7 committed:
Move analytics info into CourseInfoBar

EricPedley reviewed (changes requested):
Try it out with this extension (https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) on and check the console. I don't see the analytics event firing for some reason. It fires a generic "click" event but not the custom "Click Prerequisites" event.

Voark committed:
Staging Action Improvements (#433)

* Add action trigger on label/unlabel

This makes it so adding the "no deploy" label will destroy the stack if it is previously destroyed and deploys the stack if the "no deploy" label is removed.

* Change from npx cdk to aws to destroy stack

Instead of building the cdk and using npx cdk to destroy the stack we can just use the aws command which simplifies many of the build steps.

* Replace Github deployment with URL comment

Instead of dealing with the permission difficulties that come with Github deployments we instead will just use an action that comments the staging URL, and recreates on update.

* Hard code AWS credentials

Instead of using OIDC to provide temporary credentials we will use AWS credentials stored in the repo secrets. With OIDC, there will not be a good way to get AWS credentials when pull requests come from outside the repo.

* Add concurrency and wait for stack status

Because there is no reason for multiple instances of the action to run at once we add concurrency to force only one instance at a time. That being said, when cancelling another workflow instance sometimes the stack may not have finished updating so adding stack status checks ensures the action will wait until the stack is done updating before attempting more changes.

EricPedley committed:
changed 'positions' to 'opportunities' in recruitment banner (#441)

EricPedley committed:
add 2023 winter to term data

ChaseC99 committed:
Fix month indexing for Winter '23 (#445)

* Fix month indexing for Winter '23

Months are 0-indexed, so January should `0` instead of `1`

* Update src/termData.js

EricPedley committed:
fix event in too many schedules bug (#446)

EricPedley committed:
Add contributor video tutorial link

Voark committed:
Fix Github Actions not checking status properly (#447)

* Fix actions stack status query

* Move stabilize stack to reusable action

* add staging environmental variable

EricPedley committed:
Update action.yaml (#451)

EricPedley committed:
Fix duplicate section code bug (#453)

* make getCourseData accumulator work for duplicate section codes

* remove proxy URL

* add separate error case for 404s on user ID

ryanmohta committed:
Fix broken review links involving special characters (#455)

JacE070 committed:
Remove Google Analytics v3 (#459)

* Removed all ReactGA imports and function calls

* Removed redundant code

* Removed react-ga from package.json

EricPedley committed:
Convert codebase to TypeScript (#411)

MinhxNguyen7 committed:
Duplicate section meetings from different locations (#465)

JacE070 committed:
Fixed GE-III Bug (#463)

JacE070 committed:
Missing waypoints for course section with fake locations (#471)

* Filtered out waypoints with fake locations

* removed folder accidentally added

JacE070 committed:
Add Spring 2023 Class Schedule (#480)

ap0nia committed:
simplified copy to clipboard (#479)

- why would you make a new DOM element to write a raw value to the clipboard????????????????????????????????????????
- caveat: it only works in "secure contexts", e.g. localhost and production website
- https://stackoverflow.com/questions/51805395/navigator-clipboard-is-undefined

JacE070 committed:
Remove excessive toggleSearch() (#484)

ap0nia committed:
Vite yarn (#486)

* switched from npm to yarn

* added vite and config

* moved index html to root and index.js to jsx

* changed require to import

* added vite commands to package.json

JacE070 committed:
Dark Mode Screenshots Bug (#462)

* Change screenshot background color to corresponding theme

* Fixed dark mode screenshots bug

* Remove unnecessary  style modifications

* Remove CSS styles not required

* Remove codes dealing with color in CalendarRoot

* Remove unused import

Voark committed:
Refactor schedules data structure (#457)

* Reimplement delete schedule functionality

* Reimplement clear schedule functionality

* Reimplement change color functionality

* Reimplement basic custom event functionality

* Reimplement custom event change color functionality

* Minor refactoring (mainly to adding courses)

- move logic related to add courses out of AppStoreActions
- change Schedules.AddCourse to use existing course if exists, and if not assign color
- fix addSchedule to change current schedule number to newest created schedule
- remove color argument from AppStoreActions.AddCourse because it is always undefined

* Remove redundant color property

* Add undo functionality and expand to all actions that change schedule

* Reimplement copy schedule functionality
- Also fix bug with adding course to not current schedule index

* Reimplement load and save functionality for new schedules

* Linting and merge fixes

* Fix schedule clear not clearing custom events

* Refactor loadSchedule

* Load schedules more efficiently with some minor refactors
- Load courses a term at a time instead of one course at a time

* Move schedule types to separate file

* Add documentation to Schedules and reorganize methods

* Fix more merge issues

* Minor color pick logic refactor

Co-authored-by: Aponia Enjoyer <brianmv@uci.edu>

* Remove extra import

* Calculate calenderized events on demand
- Also renamed calendarizeHelpers.ts because it was bothering me

* Change calendarize functions to get courses as arguments
- Also added documentation to calendarize functions in Schedule

* Change calendarize helpers to explicit typing

* Clarify comment

* Update comments in Schedules

---------

Co-authored-by: Aponia Enjoyer <brianmv@uci.edu>

Voark committed:
Fix schedule loading bugs (#505)

- Bug 1: Querying websoc with more than 10 courses at once
- Bug 2: Querying websoc for cancelled course

ap0nia committed:
Fix null (#506)

* fixed bug that kills the app on start

- if you click around the tabs while the schedule is loading, everything
  dies :rocket: :skull:

* typo in map

Voark committed:
Fix staging (#504)

* Fix staging API url

* Add package lock and vite config to eslint ignore

Co-authored-by: senti <brianmv@uci.edu>

* Restore accidentally deleted file

* Clean up vite-env

---------

Co-authored-by: senti <brianmv@uci.edu>

mina03333 committed:
Schedule Help Box Removed (#475)

* Schedule Help Box Removed

* Removed Schedule Help Box

* make variable naming clearer

---------

Co-authored-by: Mina Yoon <jungmy1@uci.edu>
Co-authored-by: Eric Pedley <ericpedley@gmail.com>

ap0nia committed:
Refactored Root Structure + Aliases + Future Prep (#493)

* added updated packages

- merged all packages that my re-write aponia branch needs
- all functionality of the site is exactly the same

- this allows me to work on implementing the overhaul without continuously
  changing `package.json` and `yarn.lock`

* added aliases to the project

* converted prettier and eslint config to yaml

- they're a lot easier to understand or parse at a glance IMO

* added types for react-input-mask

- it kills `tsc` without it for some reason

* Refactor root structure (#494)

* moved index.jsx to index.tsx

- now it's typesafe :sunglasses:
- renamed the script import in index.html to match

* moved App to the root, next to index.tsx

* moved the theme to its own custom provider

* converted App to a function component

* moved react-app-env.d.ts to vite-env.d.ts

* added query provider

* simplified index.tsx by moving theme to provider

* forgot to add snackbar provider

* adjusted the main colors

* Refactor aliases (#495)

* moved index.jsx to index.tsx

- now it's typesafe :sunglasses:
- renamed the script import in index.html to match

* moved App to the root, next to index.tsx

* moved the theme to its own custom provider

* converted App to a function component

* moved react-app-env.d.ts to vite-env.d.ts

* added lib directory and moved root files there

- I prefer the entry point folder with App to be unclutterred

* fixed aliasing in stores

* moved api/endpoints to lib

* updated aliases in actions

* updated aliases in theme provider

* updated aliases in lib helpers

* updated aliases in AppBar components

* updated imports in Home and MobileHome

* updated aliases in top level calendar

* updated aliases in top level of calendar toolbar

* updated aliases in sublevels of calendar toolbar

* updated aliases in root of right pane

* updated aliases in added courses

* updated aliases in root of course pane

* updated aliases for search form

* updated aliases for map

* updated aliases in section table

* removed the duplicate files that are now in lib

* fixed imports in color picker

* fixed import statements in schedule store

* update lockfiles

* forgot import statement to load schedule

* update yarn.lock (again)

* update docs for vite

* change npm start to yarn start

---------

Co-authored-by: Eric Pedley <ericpedley@gmail.com>

ap0nia committed:
added split pane and fixed config (#507)

* added split pane

* re-added linting

- idk why it's disabled?
- move husky and lint-staged config out of package.json (more modern
  practice?)
  - husky hooks are generally executable files in the .husky folder
  - .lintstagedrc.yml configures lint-staged comand for the project

* added custom CSS for react-split

- eslint got made at me for adding a CSS file, so I adjusted the ignore
  pattern

* updated eslint and lint staged rules

- [jsx-runtime stops eslint from complaining about not seeing `import React from 'react'`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md)
- updated lintstaged to run on jsx, tsx files and ignore html,json,css
  files

* added the ellipsis

- the strat is very dumb, so I wrote down a lot of explanations

EricPedley committed:
Run linter on entire project (#508)

* fix linting problems

* lint .ts files

* remove pages from format command

* run prettier on entire project (minus UCIMap)

* add formatter to pre-commit hook

* remove formatter from pre-commit hook

teresa-liang committed:
Default class colors (#476)

* have sections of the same course be the same hue but different shades

* have sections of the same course be the same color

* fix linting problems

* lint .ts files

* re-implement color changes in Schedules.ts

* remove unused hex color helper functions

---------

Co-authored-by: Eric Pedley <ericpedley@gmail.com>

johnlorenzini committed:
Patchnotes modal (#481)

* awaiting content

* added content to dialog

* added content to dialog fix

* readability fixes

* updated patch notes for recent changes

---------

Co-authored-by: Eric Pedley <ericpedley@gmail.com>

ap0nia committed:
fix mobile split pane bug (#511)

* oops killed mobile

* managed to preserve the ellipsis icon

EricPedley committed:
replace structuredClone with json parse fuckery (#513)

EricPedley committed:
Fix ColorPicker not showing section color right after adding. (#512)

* make addCourse return a color and use it in sectiontablebuttons

* make addCourse return the added course instead of just the color

* add temp variable for clarity

JacE070 committed:
Fix bug by stripping string for condition checking (#517)

JacE070 committed:
Add note for Map tba bug (#519)

* Fix bug by stripping string for condition checking

* Delete redundant import

* Add note for trimming operation

---------

Co-authored-by: Aponia Enjoyer <brianmv@uci.edu>

MinhxNguyen7 committed:
Merge branch 'main' into move-prereqs-link

EricPedley committed:
Merge branch 'main' into move-prereqs-link

EricPedley reviewed (approved):
LGTM gonna merge blind without the preview 😎
